pkg/kf/cli/prompts.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/manifoldco/promptui"
+)
+
+type interactiveModeType struct{}
+
+// SetInteractiveMode returns a context that marks it as interactive. This
+// allows prompts to make decisions based on if the user wants to interact
+// with the CLI interactively or not.
+func SetInteractiveMode(ctx context.Context, interactive bool) context.Context {
+	return context.WithValue(ctx, interactiveModeType{}, interactive)
+}
+
+// GetInteractiveMode returns true if SetInteractiveMode has been set to true
+// for the given context.
+func GetInteractiveMode(ctx context.Context) bool {
+	i, _ := ctx.Value(interactiveModeType{}).(bool)
+	return i
+}
+
+// interactiveModeDisabledErr is returned by a prompt if InteractiveMode is
+// not enabled but a prompt is still requested.
+type interactiveModeDisabledErr struct {
```

Comment:
I think it makes sense for this to be a declared error instead `var IntermactiveModeDisabledErr = errors.New("prompt requested but interactive mode is disabled")`

Reply:
I started with this, but I wanted to be able to offer more information in the error (e.g., what failed)